### PR TITLE
[SW-2298] Update mojo pipeline doc

### PR DESCRIPTION
**Pull request by jakubhava**
The initial motivation for this change was to add troubleshooting section, but I realized that the documentation could use a little bit of update

### CODE REVIEW

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+        .. code:: python
+
+            predictions.select("prediction")
+
+The output data frame contains all the original columns plus the prediction column which is by default named
+``prediction``. The prediction columns contains all the prediction detail. Its name can be modified via the ``H2OMOJOSettings``
+object.
+
+Customizing the MOJO Settings
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:143]: Have you changed anything above this section?

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can configure the output and format of predictions via the H2OMOJOSettings. The available options are
+
+- ``predictionCol`` - Specifies the name of the generated prediction column. Default value is `prediction`
+- ``detailedPredictionCol`` - Specified the name of the generated detailed prediction column. The detailed prediction column,
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:149]: Are all of the properties relevant to the pipeline mojo? E.g. `withContributions`, `withDetailedPredictionCol`, `withDetailedPredictionCol `

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 1 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:143]: It's completely rewritten but the message remains the same

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 2 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:149]: Good point, will verify which ones are related

Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: Thanks, this was really good catch. The valid options are only:

`predictionCol`, `convertUnknownCategoricalLevelsToNa`, `convertInvalidNumbersToNa` and `namedMojoOutputColumns`

File: doc/src/site/sphinx/deployment/load_mojo.rst [hunk 3]
```diff
@@ -247,6 +247,21 @@ and in case regression problem the predicted number. If we need to access more d
...
 
+Customizing the MOJO Settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can configure the output and format of predictions via the H2OMOJOSettings. The available options are
+
+- ``predictionCol`` - Specifies the name of the generated prediction column. Default value is `prediction`
+- ``detailedPredictionCol`` - Specified the name of the generated detailed prediction column. The detailed prediction column,
+  if enabled, contains additional details such as probabilities, shapley values etc. The default value is `detailed_prediction`.
+- ``withDetailedPredictionCol`` - Enables or disables generation of detailed prediction column. It is disabled by default.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo.rst:258]: `detailed prediction column` -> `the detailed prediction column`

File: doc/src/site/sphinx/deployment/load_mojo.rst [hunk 4]
```diff
@@ -247,6 +247,21 @@ and in case regression problem the predicted number. If we need to access more d
...
+- ``convertUnknownCategoricalLevelsToNa`` - Enables or disables conversion of unseen categoricals to NAs. By default it is disabled.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo.rst:259]: `By default it is disabled.` -> `By default, it is disabled.`

File: doc/src/site/sphinx/deployment/load_mojo.rst [hunk 5]
```diff
@@ -247,6 +247,21 @@ and in case regression problem the predicted number. If we need to access more d
...
+- ``convertInvalidNumbersToNa`` - Enables or disables conversion of invalid numbers to NAs. By default it is disabled.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo.rst:260]: ditto

File: doc/src/site/sphinx/deployment/load_mojo.rst [hunk 6]
```diff
@@ -247,6 +247,21 @@ and in case regression problem the predicted number. If we need to access more d
...
+- ``withContributions`` - Enables or disables computing shapley values. Shapley values are generated as a sub-column for the de
+  detailed prediction column. Therefore to obtain shapley value, both this option and ``withDetailedPredictionCol`` needs to be
+  enabled. By default it is disabled.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo.rst:263]: ditto

File: doc/src/site/sphinx/deployment/load_mojo.rst [hunk 7]
```diff
@@ -247,6 +247,21 @@ and in case regression problem the predicted number. If we need to access more d
 additional column will be created during predictions, by default named ``detailed_prediction`` which contains, for example,
 predicted probabilities for each predicted label in case of classification problem, shapley values and other information.
 
+Customizing the MOJO Settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can configure the output and format of predictions via the H2OMOJOSettings. The available options are
+
+- ``predictionCol`` - Specifies the name of the generated prediction column. Default value is `prediction`
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo.rst:255]: `prediction` -> `prediction.`

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 8]
```diff
@@ -0,0 +1,161 @@
+Importing MOJO Pipelines from Driverless AI
+-------------------------------------------
+
+MOJO scoring pipeline artifacts, created in Driverless AI, can be used in Spark to carry out predictions in parallel
+using the Sparkling Water API. This section shows how to load and run predictions on the MOJO scoring pipeline in
+Sparkling Water
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:6]: `Sparkling Water` -> `Sparkling Water.`

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        .. code:: bash
+
+            ./bin/pyspark --jars license.sig --py-files py/build/dist/h2o_pysparkling_SUBST_SPARK_MAJOR_VERSION-SUBST_SW_VERSION.zip
+
+        .. code:: bash
+
+            ./bin/pysparkling --jars license.sig
+
+
+At this point, you should have Spark interactive terminal where we can carry out predictions.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:57]: `you` vs. `we` in the paragraph

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+.. content-tabs::
+
+    .. tab-container:: Scala
+        :title: Scala
+
+        .. code:: scala
+
+            import ai.h2o.sparkling.ml.models.H2OMOJOPipelineModel
+            val settings = H2OMOJOSettings()
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:71]: What about using some non default values of `H2OMOJOSettings` in the example.

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+Prepare the dataset to score on:
+
+.. content-tabs::
+
+    .. tab-container:: Scala
+        :title: Scala
+
+        .. code:: scala
+
+            val dataFrame = spark.read.option("header", "true").csv("file:///path/to/the/data.csv")
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:94]: Does it make a difference if schema is inferred? In other words, are values converted to string before passing them to a pipeline model?

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+Customizing the MOJO Settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can configure the output and format of predictions via the H2OMOJOSettings. The available options are
+
+- ``predictionCol`` - Specifies the name of the generated prediction column. Default value is `prediction`
+- ``convertUnknownCategoricalLevelsToNa`` - Enables or disables conversion of unseen categoricals to NAs. By default it is disabled.
+- ``convertInvalidNumbersToNa`` - Enables or disables conversion of invalid numbers to NAs. By default it is disabled.
+- ``namedMojoOutputColumns`` - Enables or disables named output columns. When enabled, the ``predictionCol`` contains sub-columns
+  with names corresponding the the labels we try to predict. If disables, the ``predictionCol`` contains the array of predictions without
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:152]: `If disables` -> `If disabled`

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+``prediction``. The prediction columns contains all the prediction detail. Its name can be modified via the ``H2OMOJOSettings``
+object.
+
+Customizing the MOJO Settings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We can configure the output and format of predictions via the H2OMOJOSettings. The available options are
+
+- ``predictionCol`` - Specifies the name of the generated prediction column. Default value is `prediction`
+- ``convertUnknownCategoricalLevelsToNa`` - Enables or disables conversion of unseen categoricals to NAs. By default it is disabled.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:149]: ditto: `By default` ->  `By default,`

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+- ``convertInvalidNumbersToNa`` - Enables or disables conversion of invalid numbers to NAs. By default it is disabled.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:150]: ditto

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+- ``namedMojoOutputColumns`` - Enables or disables named output columns. When enabled, the ``predictionCol`` contains sub-columns
+  with names corresponding the the labels we try to predict. If disables, the ``predictionCol`` contains the array of predictions without
+  the column names. By default it is enabled.
```
Comment (mn-mikke) [inline doc/src/site/sphinx/deployment/load_mojo_pipeline.rst:153]: ditto

File: doc/src/site/sphinx/deployment/load_mojo.rst (hunk 5 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo.rst (hunk 4 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo.rst (hunk 3 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo.rst (hunk 7 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo.rst (hunk 6 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 8 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 13 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 14 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 12 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 15 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: 👍

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 9 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: Thanks!

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 11 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: Good questions. Actually both modes are supported.

If you pass MOJOPipeline all strings, it parses the column to the appropriate type of the column generated by DAI. If you provide the types,  our code does not need to do the parse -> faster.

I will add the inferSchema code to the example. Thanks!

File: doc/src/site/sphinx/deployment/load_mojo_pipeline.rst (hunk 10 above)
Comment (jakubhava) [reply doc/src/site/sphinx/deployment/load_mojo_pipeline.rst]: Good idea!

**Pull request merged**